plenum/server/view_change/pre_view_change_strategies.py

```diff
@@ -0,0 +1,72 @@
+from abc import abstractmethod, ABCMeta
+from functools import partial
+
+from plenum.common.messages.fields import IntegerField
+from plenum.common.messages.message_base import MessageBase
+from plenum.common.constants import VIEW_CHANGE_START, PROPOSED_VIEW_NO, PreVCStrategies
+
+
+class ViewChangeStartMessage(MessageBase):
+    typename = VIEW_CHANGE_START
+    schema = (
+        (PROPOSED_VIEW_NO, IntegerField()),
+    )
+
+
+class PreViewChangeStrategy(metaclass=ABCMeta):
+    """Abstract class for routines before starting viewChange procedure"""
+
+    def __init__(self, view_changer):
+        self.view_changer = view_changer
+
+    @abstractmethod
+    def vc_preparation(self, proposed_view_no: int):
+        raise NotImplementedError()
+
+
+class VCStartMsgStrategy(PreViewChangeStrategy):
+    """Strategy logic:
+    - when startViewChange method was called, then put 'local' ViewChangeStart message
+    - when this message will be processed by node's nodeInBoxRouter then it will be added to replica's inBox queue
+    - when this message will be processed by replica's inBoxRouter then startViewChange method will be called
+      and view_change procedure will be continued
+    """
+
+    def vc_preparation(self, proposed_view_no: int):
+        self.set_req_handler()
+        vcs_msg = ViewChangeStartMessage(proposed_view_no)
+        nodeInBox = self.view_changer.node.nodeInBox
+        nodeInBox.append((vcs_msg, self.view_changer.node.name))
```

Comment:
It means that we will process only a quota of nodestack messages before we process  ViewChangeStartMessage during the next loop. I think we should try to process more than a standard quota (quota X 10)?

Reply:
All messages from nodestack will be placed AFTER  ViewChangeStartMessage, so the strategy will not have any effect, won't it?